File: app/src/main/java/com/mancode/financetracker/database/pojos/TransactionFull.java
package com.mancode.financetracker.database.pojos;

import com.mancode.financetracker.database.converter.DateConverter;
import com.mancode.financetracker.database.entity.TransactionEntity;

import org.threeten.bp.LocalDate;

import androidx.room.ColumnInfo;
import androidx.room.Embedded;
import androidx.room.TypeConverters;

@TypeConverters(DateConverter.class)
public class TransactionFull {

    @Embedded
    public TransactionEntity transaction;
    @ColumnInfo(name = "account_name")
    public String accountName;
    @ColumnInfo(name = "account_type")
    public int accountType;
    @ColumnInfo(name = "account_currency")
    public String currency;
    @ColumnInfo(name = "account_open_date")
    public LocalDate openDate;
    @ColumnInfo(name = "account_close_date")
    public LocalDate closeDate;
    @ColumnInfo(name = "category")
    public String category;
    @ColumnInfo(name = "category_type")
    public int categoryType;

    public TransactionFull(TransactionEntity transaction,
                           String accountName,
                           int accountType,
                           String currency,
                           LocalDate openDate,
                           LocalDate closeDate,
                           String category,
                           int categoryType) {
        this.transaction = transaction;
        this.accountName = accountName;
        this.accountType = accountType;
        this.currency = currency;
        this.openDate = openDate;
        this.closeDate = closeDate;
        this.category = category;
        this.categoryType = categoryType;
    }
}

File: app/src/main/java/com/mancode/financetracker/database/pojos/TransactionFull.kt
package com.mancode.financetracker.database.pojos

import androidx.room.ColumnInfo
import androidx.room.Embedded
import androidx.room.TypeConverters
import com.mancode.financetracker.database.converter.DateConverter
import com.mancode.financetracker.database.entity.TransactionEntity
import org.threeten.bp.LocalDate

@TypeConverters(DateConverter::class)
class TransactionFull(@field:Embedded
                      var transaction: TransactionEntity,
                      @field:ColumnInfo(name = "account_name")
                      var accountName: String,
                      @field:ColumnInfo(name = "account_type")
                      var accountType: Int,
                      @field:ColumnInfo(name = "account_currency")
                      var currency: String,
                      @field:ColumnInfo(name = "account_open_date")
                      var openDate: LocalDate,
                      @field:ColumnInfo(name = "account_close_date")
                      var closeDate: LocalDate,
                      @field:ColumnInfo(name = "category")
                      var category: String,
                      @field:ColumnInfo(name = "category_type")
                      var categoryType: Int)
